core/federated/RTI/rti_common.c

```diff
@@ -115,6 +117,7 @@ tag_advance_grant_t tag_advance_grant_if_safe(scheduling_node_t* e) {
     // To handle cycles, need to create a boolean array to keep
     // track of which upstream enclave have been visited.
     bool *visited = (bool *)calloc(rti_common->number_of_scheduling_nodes, sizeof(bool)); // Initializes to 0.
+    visited[e->id] = true;
```

Comment:
This and the corresponding change in `rti_remote.c` seems to cause a test failure in `FeedbackDelay.lf`. Or, at least, that tests passes on the original `enclaves2` branch. I dont see this being done on `main` either?